frappe/public/js/frappe/model/model.js

```diff
@@ -103,20 +103,47 @@ $.extend(frappe.model, {
 		return docfield[0];
 	},
 
+	get_from_localstorage: function(doctype) {
+		if (localStorage["_doctype:" + doctype]) {
+			return JSON.parse(localStorage["_doctype:" + doctype]);
+		}
+	},
+
+	set_in_localstorage: function(doctype, docs) {
+		try {
+			localStorage["_doctype:" + doctype] = JSON.stringify(docs);
+		} catch(e) {
+			// if quota is exceeded, clear local storage and set item
+			console.warn("localStorage quota exceeded, clearing doctype cache")
+			frappe.model.clear_local_storage();
+			localStorage["_doctype:" + doctype] = JSON.stringify(docs);
+		}
+	},
+
+	clear_local_storage: function() {
+		for(var key in localStorage) {
+			if (key.startsWith("_doctype:")) {
```

Comment:
Can we use https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API instead of clearing cached data?

Reply:
We can introduce [this library](https://github.com/localForage/localForage) if required.

Reply:
Yeah, seems like a good idea